### ISSUE TITLE
RePaintPipeline always returns black image when pipe.to("mps") is called

### ISSUE THREAD

**Issue by GaidamakUA** [huggingface/diffusers#1251]
### Describe the bug

When I try to do the inpainting with RePaintPipeline on the MPS backend it only returns black image. It works fine with CPU, but very slow.

### Reproduction

```python
scheduler = RePaintScheduler.from_config("google/ddpm-ema-celebahq-256")
pipe = RePaintPipeline.from_pretrained("google/ddpm-ema-celebahq-256", scheduler=scheduler)
device = torch.device("mps")
pipe = pipe.to(device)
def inpaint(image, mask):
    output = pipe(
        original_image=image,
        mask_image=mask,
        num_inference_steps=20,
        eta=0.0,
        jump_length=1,
        jump_n_sample=2,
        return_dict=False,
    )
    print(output)
    image.paste(output[0][0], (x, y))
    return np.array(image)
```

### System Info

- `diffusers` version: 0.7.2
- Platform: macOS-12.6-arm64-arm-64bit
- Python version: 3.9.12
- PyTorch version (GPU?): 1.13.0.dev20221002 (False)
- Huggingface_hub version: 0.10.0
- Transformers version: 4.24.0
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no

**Comment by patrickvonplaten** [comment]
Interesting! We did some bigger changes for cycle diffusion on `"main"` - could you maybe try again?

**Comment by patrickvonplaten** [comment]
Also maybe cc @pcuenca

**Comment by pcuenca** [comment]
Hi @GaidamakUA, thanks for reporting this.

It doesn't work for me either, but it does not return black images. This is what I get on `mps` for the image and mask taken from [this example](https://huggingface.co/docs/diffusers/api/pipelines/repaint):

![repaint](https://user-images.githubusercontent.com/1177582/202421697-0e5e2054-efca-4bfa-b1d7-0bbab7225d4b.png)

For reference, this is the result when running on `cuda` (not perfect, I just ran it for a few steps):

![repaint_cuda](https://user-images.githubusercontent.com/1177582/202421761-cc62e272-49f0-493e-9563-94fc8dfccffb.png)

What's also puzzling is that the unmasked image area to be preserved (the bottom half of the face) appears to have inverted colors in the `mps` version.

Is this the same thing you are seeing? Also, do you know if this ever worked in a previous version of diffusers?

**Comment by GaidamakUA** [comment]
Still black
- `diffusers` version: 0.8.0.dev0
- Platform: macOS-12.6-arm64-arm-64bit
- Python version: 3.9.12
- PyTorch version (GPU?): 1.13.0.dev20221002 (False)
- Huggingface_hub version: 0.10.0
- Transformers version: 4.24.0
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>
Input
![in](https://user-images.githubusercontent.com/2983027/202801071-52bbee3c-e24d-41b6-af5e-2b6c8711ef92.png)
Mask
![mask](https://user-images.githubusercontent.com/2983027/202801109-877917f3-6c0b-4406-be21-ac12e6eaa9c8.png)
Output
![out](https://user-images.githubusercontent.com/2983027/202801488-048e56f3-27c8-44ac-a4e8-d86cb550d9de.png)

**Comment by GaidamakUA** [comment]
@pcuenca any chances for a fix?

**Comment by patrickvonplaten** [comment]
@anton-l could you take a look?

**Comment by patrickvonplaten** [comment]
Gently ping @pcuenca @anton-l, but also please note @GaidamakUA that our MPS integration is still very much experimental as it was only recently supported in PyTorch and RePaint is also not the most used pipeline I believe :sweat_smile:

**Comment by pcuenca** [comment]
Yeah, it's in my to-do list but I still haven't found the time to dive deeper after the initial assessment I made when @GaidamakUA reported the problem. Unfortunately I'm fully booked up until the 15th, will come back to this then. Sorry about that!

**Comment by pcuenca** [comment]
Maybe related: https://github.com/invoke-ai/InvokeAI/commit/bd57793a658a86d5d81f86401e2d90ec10a92046, https://github.com/pytorch/pytorch/issues/91617, https://github.com/kulinseth/pytorch/pull/222
